xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodec.cpp

```diff
@@ -116,8 +116,8 @@ CRenderInfo CRendererMediaCodec::GetRenderInfo()
 {
   CRenderInfo info;
   info.formats = m_formats;
-  info.max_buffer_size = NUM_BUFFERS;
-  info.optimal_buffer_size = 2;
+  info.max_buffer_size = 4;
```

Comment:
> not setting optimal_buffer_size results in actual buffers used = 2

Actually, I understand from https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp#L284 that its the lowest value between the codec requirements, the renderer requirements (optimal and max) and "6".
If nothing is implemented codec and hw-rendrer wise, the value would be 4, from CLinuxRendererGLES, but the absolute minimum is indeed "2".

De facto, "2" causes problems after a refresh rate change, but is fine otherwise. That might point out to some deadlocked buffers in the render loop after a refresh change, though, so a different root cause.